### ISSUE TITLE
05-random-forest-diabetes.ipynb

### ISSUE THREAD

**Issue by mezcolantriz** [4GeeksAcademy/marcoakai_machine_learning#8]
Bien @Marcoakai , te digo más abajo qué modificar para dejarlo perfecto.

### Lo que está bien:

1. **Imports correctos** - Incluye todo lo necesario 

2. **Split correcto** (aunque falta `stratify=y`)

3. **Modelo base para comparación** - RandomForest con n_estimators=100 

4. **Búsqueda manual de hiperparámetros:**
   - Prueba `n_estimators`: [10, 50, 100, 200]
   - Prueba `max_depth`: [3, 5, 7, None]
   - Guarda resultados en lista 

5. **Visualización de resultados:**
   - Gráfica del impacto de hiperparámetros 
   - Muy buena práctica

6. **Mejor modelo seleccionado** (n_estimators=100, max_depth=5)

7. **Evaluación en train y test** - Para detectar overfitting 

8. **Visualización de árboles del bosque** - Muy buen detalle 

9. **Modelo guardado con pickle** 

### **Mejoras necesarias:**

1. **Falta `stratify=y`** en el split ya que las clases están desbalanceadas:
   ```python
   X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42, stratify=y)
   ```

2. **Métricas incompletas** - Solo accuracy, falta:
   - Precision, Recall, F1-score
   - Confusion matrix
   - Classification report
  Con su debida interpretación por escrita :) 

3. **Usar GridSearchCV en vez de bucles manuales** sería más elegante
